src/ru/javarush/borets/module2/lifeCycle/Viable.java

```diff
@@ -0,0 +1,108 @@
+                    }
+                }
+                AnimalFactory factory = new AnimalFactory();
+                Map<String, Long> result = listForReproduce.stream()
+                        .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+
+                for (Map.Entry<String, Long> item : result.entrySet()) {
+                    int countPossibleChild = (int) (item.getValue() / NUMBER_OF_PARENTS);
+                    int differenceBetweenMaxAndPossible = (int) (maxQuantity.get(Entities.valueOf(item.getKey().toUpperCase())) - item.getValue());
+
+                    for (int i = 0; i < Math.random() * countPossibleChild && i < differenceBetweenMaxAndPossible; i++) {
+                        gameField[row][column].add(factory.getAlive(Entities.valueOf(item.getKey().toUpperCase())));
+                        countOfReproduce.getAndIncrement();
+                    }
+                }
+            }
+        }
+        report.append(countOfReproduce).append(" animals were born " + "\n");
+    }
+
+    static void eat() throws FileNotFoundException {
+        for (int row = 1; row < gameField.length - 1; row++) {
+            for (int column = 1; column < gameField[0].length - 1; column++) {
+                ArrayList<Alive> tempArray = gameField[row][column];
+                for (int i = 0; i < tempArray.size(); i++) {
+                    if (tempArray.get(i) instanceof Caterpillar caterpillar) {
+                        caterpillar.caterpillarEat(i, row, column);
+                    } else if ((tempArray.get(i) instanceof Animal animal) && (!animal.isAte)) {
+                        if (animal instanceof Herbivores herbivores) {
+                            herbivores.eatPlant(animal, i, row, column);
+                        } else if (animal instanceof Predator predator) {
+                            predator.eat(animal, i, row, column);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    static void moveToAnotherLocation() {
+        //Direction direction = randomDirectionMove();
+        AtomicInteger count = new AtomicInteger(0);
+        for (int row = 1; row < gameField.length - 1; row++) {
+            for (int column = 1; column < gameField[0].length - 1; column++) {
+                ArrayList<Alive> tempArray = gameField[row][column];
+
+                for (int i = 0; i < tempArray.size(); i++) {
+                    if (tempArray.get(i) instanceof Activity) {
+                        log.info("  Animal moved " + tempArray.get(i));
+                        ((Activity) tempArray.get(i)).move(row, column, tempArray.get(i), randomDirectionMove());
+                        count.getAndIncrement();
+                    }
+                }
+            }
+        }
+        report.append(count).append(" animals moved " + "\n");
+    }
+
+    private static Direction randomDirectionMove() {
```

Comment:
это наверное по логике метод 
```suggestion
    private static Direction getRandomDirectionMove() {
```